.github/workflows/pr-builder.yml

```diff
@@ -0,0 +1,163 @@
+# -------------------------------------------------------------------------------------
+#
+# Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+#
+# WSO2 LLC. licenses this file to you under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# --------------------------------------------------------------------------------------
+
+# This workflow will run the build for PRs to check for any build related issues.
+
+name: 👷 PR Builder
+
+on:
+  pull_request:
+    types: [ opened, synchronize, labeled ]
+    branches: [ main ]
+    paths-ignore:
+      - "**/LICENSE"
+
+concurrency:
+  group: pr-builder-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  GH_TOKEN: ${{ secrets.RELEASE_BOT_TOKEN }}
+  ANDROID_SCRIPT_DIR: .github/workflows/scripts/android
+  DOCS_SCRIPT_DIR: .github/workflows/scripts/docs
+  DOC_DIR: docs
+
+jobs:
+  check_for_relevant_changes:
+    name: 🔄 Check for Relevant Changes
+    runs-on: ubuntu-latest
+    outputs:
+      android_changed: ${{ steps.check_android_changes.outputs.android_changed }}
+      documentation_changed: ${{ steps.check_documentation_changes.outputs.documentation_changed }}
+    steps:
+    - name: ⬇️ Checkout
+      uses: actions/checkout@v3
+      with:
+        token: ${{ env.GH_TOKEN }}
+        fetch-depth: 0
+
+    - name: 🌴 Fetch PR branch
+      run: |
+        if [ ${{ github.event.pull_request.head.repo.fork }} == true ]; then
+          git fetch origin pull/${{ github.event.pull_request.number }}/head:pr-${{ github.event.pull_request.number }}
+        else
+          git fetch origin ${{ github.head_ref }}
+        fi
+
+    - name: 🗃️ Get changed files
+      id: changed-files
+      run: |
+        if [ ${{ github.event.pull_request.head.repo.fork }} == true ]; then
+          echo "Comparing changes for PR from fork"
+          git diff --name-only origin/${{ github.base_ref }}..pr-${{ github.event.pull_request.number }} > changed_files.txt
+        else
+          echo "Comparing changes for internal PR"
```

Comment:
Internal pr sounds like a private contribution IMO. shall we rename/remove this log?

PS: can't we get the diff using the pr number in both cases?

Reply:
fixed with https://github.com/asgardeo/mobile-ui-sdks/pull/49/commits/869590ee84080b7793cad437f8c452f5ec3f01aa